src/app/admin/add-author/page.jsx

```diff
@@ -0,0 +1,100 @@
+'use client'
+
+import React, { useState } from 'react'
+import { useMutation } from '@apollo/client'
+import { CREATE_AUTHOR } from '../../graphql/queries'
+
+export default function AddAuthor() {
+  const [name, setName] = useState('')
+  const [biography, setBiography] = useState('')
+  const [bornDate, setBornDate] = useState('')
+  const [profileImageUri, setProfileImageUri] = useState('')
+  const [errorMessage, setErrorMessage] = useState('')
+
+  const [createAuthor, { loading: createAuthorLoading }] = useMutation(CREATE_AUTHOR)
+
+  const handleSubmit = async (e) => {
+    e.preventDefault()
+    setErrorMessage('')
+    try {
+      await createAuthor({
+        variables: { name, biography, bornDate, profileImageUri }
+      })
+
+      // Clear input fields on successful response
+      setName('')
+      setBiography('')
+      setBornDate('')
+      setProfileImageUri('')
```

Comment:
think of using a Form component from tpl which can help you do this + not needing to maintain state of each attribute.

Reply:
understood